applications/GeoMechanicsApplication/custom_elements/compressibility_calculator.h

```diff
@@ -0,0 +1,98 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#pragma once
+
+#include "calculator.h"
+#include "custom_retention/retention_law.h"
+#include "includes/properties.h"
+#include "includes/ublas_interface.h"
+
+#include <utility>
+#include <vector>
+
+namespace Kratos
+{
+
+class CompressibilityCalculator : public Calculator
+{
+public:
+    class InputProvider
+    {
+    public:
+        InputProvider(std::function<const Properties&()>                         rElementProperties,
+                      std::function<const std::vector<RetentionLaw::Pointer>&()> rRetentionLaws,
+                      std::function<const Matrix&()>                             rNContainer,
+                      std::function<Vector()>                        rIntegrationCoefficients,
+                      std::function<double()>                        DtPressureCoefficient,
+                      std::function<Vector(const Variable<double>&)> rNodalValuesOfDtWaterPressure)
+            : mGetElementProperties(std::move(rElementProperties)),
+              mGetRetentionLaws(std::move(rRetentionLaws)),
+              mGetNContainer(std::move(rNContainer)),
+              mGetIntegrationCoefficients(std::move(rIntegrationCoefficients)),
+              mGetDtPressureCoefficient(std::move(DtPressureCoefficient)),
+              mGetNodalValues(std::move(rNodalValuesOfDtWaterPressure))
+        {
+        }
+
+        [[nodiscard]] const Properties& GetElementProperties() const
+        {
+            return mGetElementProperties();
+        }
+
+        [[nodiscard]] const std::vector<RetentionLaw::Pointer>& GetRetentionLaws() const
+        {
+            return mGetRetentionLaws();
+        }
+
+        [[nodiscard]] const Matrix& GetNContainer() const { return mGetNContainer(); }
+
+        [[nodiscard]] Vector GetIntegrationCoefficients() const
+        {
+            return mGetIntegrationCoefficients();
+        }
+
+        [[nodiscard]] double GetDtPressureCoefficient() const
+        {
+            return mGetDtPressureCoefficient();
+        }
+
```

Comment:
See another comment, I think we should not include the time integration scheme thing here.